src/main/java/net/fabricmc/loom/configuration/MirrorConfiguration.java

```diff
@@ -0,0 +1,27 @@
+package net.fabricmc.loom.configuration;
+
+import net.fabricmc.loom.util.Constants;
+import net.fabricmc.loom.util.Mirrors;
+
+import org.gradle.api.Project;
+
+public class MirrorConfiguration {
```

Comment:
Should we make this as an extension or just a property? For users in China, if they do need to connect to say alibaba's mirror server, they currently specify it in `build.gradle`'s `repositories` block than passing in some properties.

If this is an extension named like `loomMirror`, you can still use it like:
```gradle
loomMirror {
  libraries_base = project.loom_libraries_base // a string custom base
  resources_base = project.loom_resources_base // another string custom base
  version_manifests = uri(project.version_manifests) // parsing the manifests into uri
}
```
and obtain the same behavior

Reply:
I think it makes more sense to put it in the property file or setting file instead of the build file. This thing does not change the building behavior, and hence should not be part of the build file.

Reply:
Yeah, this is fine. I don't think there is any other concerns with using project property.